lib/ids/SyncModuleIdsPlugin.js

```diff
@@ -0,0 +1,140 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+
+"use strict";
+
+const { WebpackError } = require("..");
+const { getUsedModuleIdsAndModules } = require("./IdHelpers");
+
+/** @typedef {import("../Compiler")} Compiler */
+/** @typedef {import("../Module")} Module */
+
+const plugin = "SyncModuleIdsPlugin";
+
+class SyncModuleIdsPlugin {
```

Comment:
So this should allowing to kind of map module name to id ? e.g.
```json
{
    "C:\Users\a\repo\a.css|classname_aa": "1",
    "C:\Users\a\repo\b.css|classname_aa": "2",
}
```

Reply:
Yep. Here you need to use it to sync the modules ids between Client and Server compiler so CSS Modules generate the same class names in both.

CSS Module class names are based on module ids in the new implementation.

Reply:
Example of the file: 

```
{"./style.module.css":1876,"./style2.module.css":1054}
```